Update JAMES with vwc_select function

ArjanHuizing opened the pull request:
I've added the following pathways to JAMES as we discussed

/dscore/calculate
/vwc/select

I've gone ahead and called this update 1.7.3. I did see a lot of html files which reference the old version (1.7.2), but they were so great in number I wondered if you had some sort of script that creates these.

See also my pull request for the jamesdocker repository

Commit:
Add select_vwc function to JAMES

Commit:
Update version to 1.7.3

Commit:
Update version to 1.7.3

Commit:
Merge branch 'arjan' of https://github.com/growthcharts/james into arjan

Commit:
Add passed vwc items correctly.

Commit:
Update readme

Commit:
Update select_vwc function

Commit:
Obtain dsc from time data

stefvanbuuren commented:
Excluding development packages, JAMES currently depends on 78 other packages, one of which is private.

This PR adds 36 new package(s), some of which are large. Two of them are private. 

```{r}
> size
        Package Installed_MB
1         bslib        10.68
2        cachem         0.13
3     crosstalk         1.02
4    data.table         5.85
5        digest         0.92
6     dscoreddi         4.58
7      evaluate         0.11
8        farver         4.87
9       fastmap         0.71
10  fontawesome         2.00
11      forcats         0.49
12           fs         1.52
13      ggplot2         5.88
14       gtable         0.27
15        highr         0.06
16    htmltools         0.44
17  htmlwidgets         2.00
18      isoband         2.65
19    jquerylib         1.48
20        knitr         1.42
21     labeling         0.06
22        later         2.60
23      memoise         0.06
24         mgcv           NA
25       plotly         7.13
26     promises         3.18
27     rappdirs         0.11
28 RColorBrewer         0.05
29    rmarkdown         6.53
30         sass         5.20
31       scales         0.99
32      tinytex         0.15
33  viridisLite         1.29
34          vwc         0.05
35         xfun         0.73
36         yaml         0.59
```

I will need to study how much these additional dependencies will add to the size of the application.

stefvanbuuren commented:
It seems that `plimits` is not used, so I'll remove its documentation.

```
checking Rd \usage sections ... WARNING
  Documented arguments not in \usage in Rd file 'select_vwc.Rd':
    ‘plimits’
```

stefvanbuuren commented:
I have added examples to getting started in james (dev branch) demonstrating how to call dscore/calculate endpoint. There is still a couple of things to do: 

1. Add examples of `/vwc/select` to `vignettes/articles/getting_started.Rmd`. In order to test the syntax, first build and run the docker from `jamesdocker` (`dev` branch). It would be useful to include an example where certain items are prohibited because they have already been asked.
2. The `inst/spec/openapi.in.yaml` template needs to be updated with both new end points. After some further steps, documentation will then be available from <https://james.groeidiagrammen.nl/docs/>
3. The size of the docker has grown from 2.22Gb to 2.43Gb. The only really new function is `james::select_vwc()`, which doesn't need `bslib`, `plotly`, `ggplot`, `data.table` and related packages. Any superfluous package needs to be removed from `renv.lock`.

@ArjanHuizing Would you be willing to look at these matters?

ArjanHuizing commented:
I've updated the underlying packages. As discussed, we are keeping the plotly package, despite the large amount of new dependencies it adds, as we anticipate using it for interactive plots in the future.

Additionally, I've split the `/vwc/select` function into `/vwc/select` and `/vwc/percentiles`. I've opted to return `/vwc/select` output as a whitespace (`/n`) deliminated string. The openapi yaml file has also been updated.

All my updates are on the dev branches of `james` and `jamesdocker`


I have worked on the `vignettes/articles/getting_started.Rmd` file, but had to make a few extensive changes to get bash to work on my system (Windows 11). As such I did not push this yet. Shall I put this on a separate branch?
